test/framework/namespace.go

```diff
@@ -39,27 +39,27 @@ func CreateNamespace(kubeClient kubernetes.Interface, name string) (*v1.Namespac
 	return namespace, nil
 }
 
-func (ctx *TestCtx) CreateNamespace(t *testing.T, kubeClient kubernetes.Interface) string {
+func (f *Framework) CreateNamespace(t *testing.T, ctx *TestCtx) string {
 	name := ctx.GetObjID()
-	if _, err := CreateNamespace(kubeClient, name); err != nil {
+	if _, err := CreateNamespace(f.KubeClient, f.Ctx, name); err != nil {
 		t.Fatal(err)
 	}
 
 	namespaceFinalizerFn := func() error {
-		return DeleteNamespace(kubeClient, name)
+		return DeleteNamespace(f.KubeClient, f.Ctx, name)
 	}
 
 	ctx.AddFinalizerFn(namespaceFinalizerFn)
 
 	return name
 }
 
-func DeleteNamespace(kubeClient kubernetes.Interface, name string) error {
-	return kubeClient.CoreV1().Namespaces().Delete(context.TODO(), name, metav1.DeleteOptions{})
+func DeleteNamespace(kubeClient kubernetes.Interface, ctx context.Context, name string) error {
```

Comment:
We could make it a receiver function and get rid of the first args?

```suggestion
func (f *Framework) DeleteNamespace(name string) error {
```